src/main/java/com/codesquad/qna/domain/Answer.java

```diff
@@ -0,0 +1,86 @@
+package com.codesquad.qna.domain;
+
+import com.codesquad.qna.util.DateTimeUtils;
+import org.hibernate.annotations.Where;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+@Entity
+@Where(clause = "deleted = false")
+public class Answer {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    @JoinColumn(foreignKey = @ForeignKey(name = "fk_answer_writer"))
+    private User writer;
+
+    @ManyToOne
+    @JoinColumn(foreignKey = @ForeignKey(name = "fk_answer_to_question"))
+    private Question question;
+
+    @Lob
+    private String contents;
+
+    private boolean deleted;
+
+    private LocalDateTime createdDateTime;
+
+    protected Answer() {
```

Comment:
왜 protected 기본 생성자를 생성해주셨나요?

Reply:
JPA의 엔티티에서는 기본 생성자가 무조건 필요하므로 선언해주었습니다. 그리고 클래스 외부에서 기본 생성자를 통한 인스턴스화를 막고자 private으로 선언하고 싶었으나, protected나 public만 가능하다고 JPA 표준 스펙에 명시되어있어서 그 중에서 최소한의 접근권한으로 선택하였습니다. 그렇지만 과연 상속관계에서 접근할 일이 있는건지는 모르겠습니다.

> The JPA specification requires that all persistent classes have a no-arg constructor. This constructor may be public or protected. Because the compiler automatically creates a default no-arg constructor when no other constructor is defined, only classes that define constructors must also include a no-arg constructor.

- https://openjpa.apache.org/builds/1.2.3/apache-openjpa/docs/jpa_overview_pc.html
- https://wbluke.tistory.com/6
- https://stackoverflow.com/questions/6033905/create-the-perfect-jpa-entity